### DISCUSSION TITLE
Input for trunk net in PI DeepONet

### DISCUSSION THREAD

**Discussion by voloddia** [lululxvi/deepxde#1902] [Q&A]
In all examples I've seen for PI DeepONet, input for trunk net are often coordinates of space or time or both. 
Is there any example where PDE coefficients are fed into trunk net?

**Comment by lululxvi** [comment]
You can do that. Just consider the coefficient as another dimension.

**Reply by voloddia** [reply]
Thank you, professor Lu. 
Take the [advection_aligned_pideeponet ](https://github.com/lululxvi/deepxde/tree/master/examples/operator/advection_aligned_pideeponet.py) example, if I want to add a coefficient, I guess the `pde` and `geom` will become
```
dim_x = dim_x + 1

def pde(x, y, v):
    dy_x = dde.grad.jacobian(y, x, j=0)
    dy_t = dde.grad.jacobian(y, x, j=1)
    k = x[:, 2] # not sure about index?
    return dy_t + k * dy_x

geom = dde.geometry.Rectangle([0, 0], [1, 1])
```
Is that correct? Are there other parts that need to change?

**Reply by lululxvi** [reply]
`k = x[:, 2:3]`

**Reply by voloddia** [reply]
Thank you, Professor Lu.
